docs/building-apps/esm views.md

```diff
@@ -0,0 +1,156 @@
+   dependencies are allowed to be rewritten to CDN; if not specified, its
+   default value is `**` ( -> all dependencies are rewritten)
+2. `EXTERNAL_BLOCK_LIST` is a comma-separated string that specifies which
+   dependencies are **not** allowed to be rewritten to CDN; if not specified its
+   default value is empty ( -> no dependency excluded, i.e. all dependencies are
+   rewritten)
+
+The allow / block lists are parsed and processed according to this logic:
+
+- If a dependency is only in the allow list, it will be rewritten
+- If a dependency is only in the block list, it will be bundled
+- If a dependency is in both lists, it will be bundled (`EXTERNAL_BLOCK_LIST`
+  wins)
+- If a dependency is in none of the lists, it will be bundled (but remember that
+  all dependencies are in allow list by default)
+
+The dependencies will be reflected in the output package manifest
+(`package.json`) according to these rules:
+
+- All dependencies and their versions are listed in the `dependencies` field, as
+  an object
+- The dependencies that are bundled are listed in the `bundledDependencies`
+  field, as an array
+
+It is possible to specify wildcards in the block and allow list.
+[Micromatch](https://github.com/micromatch/micromatch) syntax is supported.
+
+## Package manifest
+
+Every build of an ESM view will generate a package manifest (`package.json`),
+which will contain a selection of the original `package.json` fields, plus a set
+of added / modified fields:
+
+- [`style`](https://jaketrent.com/post/package-json-style-attribute): the
+  location of the js bundle (example:
+  `"style": "static/css/main.c6ac0a5c.css"`), useful for an host to dynamically
+  load the styles and add them to the page `<head>` or the
+  [adopted stylesheet](https://wicg.github.io/construct-stylesheets/#using-constructed-stylesheets).
+- [`module`](https://github.com/dherman/defense-of-dot-js/blob/f31319be735b21739756b87d551f6711bd7aa283/proposal.md):
+  the location of the js bundle (example:
+  `"module": "static/js/main.5077b483.js"`), useful for an host to dynamically
+  load the ESM view and render it in a React tree.
+- [`dependencies`](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#dependencies):
+  an object containing all the dependencies imported in the package source
+  (including the hoisted ones) and their correspondent version ranges.
+- [`bundledDependencies`](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#bundleddependencies):
+  an array of bundled dependencies. Since `dependencies` contains all the
+  dependencies (bundled and not bundled), it is always possible to know which
+  dependencies were rewritten and which were bundled.
+
+## External CSS imports
+
+CSS imports pointing to an external package (for example:
+[`import 'regular-table/dist/css/material.css'`](https://www.npmjs.com/package/regular-table)
+) will be rewritten to a CDN url as normal packages (for example, using skypack,
```

Comment:
URL
Skypack

"as normal packages" is unclear. What's a "normal package"?

Reply:
Yeah, "as a normal package" doesn't make sense and it's not even needed - Removed it.